packages/contracts/src/2.0.0/protocol/Exchange/libs/LibAbiEncoder.sol

```diff
@@ -207,10 +204,10 @@ contract LibAbiEncoder {
             }
 
             // Set length of calldata
-            mstore(
-                fillOrderCalldata,
-                sub(dataAreaEnd, add(fillOrderCalldata, 0x20))
-            )
+            mstore(fillOrderCalldata, sub(dataAreaEnd, add(fillOrderCalldata, 0x20)))
+
+            // Increment free memory pointer
+            mstore(0x40, add(fillOrderCalldata, dataAreaEnd))
```

Comment:
This looks off. I think it should be `mstore(0x40, dataAreaEnd)`. After the signature loop, `dataAreaEnd` will point to the first byte beyond the last one written. Adding `fillOrderCalldata` would add two pointers, which is weird.

Reply:
Whoops, that is most definitely a mistake